openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/TypeGeneratorUtils.java

```diff
@@ -233,41 +237,30 @@ public static void getRecordDocs(List<Node> documentation, Schema schemaValue,
      * @return Union type
      * @throws BallerinaOpenApiException when unsupported combination of schemas found
      */
-    public static String getUnionType(List<Schema> schemas) throws BallerinaOpenApiException {
-        StringBuilder unionType = new StringBuilder();
+    public static TypeDescriptorNode getUnionType(List<Schema> schemas) throws BallerinaOpenApiException {
+        List<TypeDescriptorNode> typeDescriptorNodes = new ArrayList<>();
         for (Schema schema: schemas) {
-            TypeGenerator typeGenerator = getTypeGenerator(schema);
-            if (!(typeGenerator instanceof RecordTypeGenerator || typeGenerator instanceof AllOfRecordTypeGenerator)) {
-                String typeName = getTypeGenerator(schema).generateTypeDescriptorNode().toString().trim();
-                if (typeName.endsWith(NILLABLE) && GeneratorMetaData.getInstance().isNullable()) {
-                    typeName = typeName.substring(0, typeName.length() - 1);
-                }
-                unionType.append(typeName);
-            } else if (typeGenerator instanceof RecordTypeGenerator) {
-                TypeDescriptorNode recordDescriptorNode;
-                if (schema.getProperties() != null) {
-                    Map<String, Schema> properties = schema.getProperties();
-                    List<String> required = schema.getRequired();
-                    List<Node> recordFList = TypeGeneratorUtils.addRecordFields(required,
-                            properties.entrySet(), false);
-                    NodeList<Node> fieldNodes = AbstractNodeFactory.createNodeList(recordFList);
-                    recordDescriptorNode = NodeFactory.createRecordTypeDescriptorNode(createToken(RECORD_KEYWORD),
-                            createToken(OPEN_BRACE_TOKEN), fieldNodes, null,
-                            createToken(CLOSE_BRACE_TOKEN));
-                } else {
-                    NodeList<Node> fieldNodes = createEmptyNodeList();
-                    recordDescriptorNode = NodeFactory.createRecordTypeDescriptorNode(createToken(RECORD_KEYWORD),
-                            createToken(OPEN_BRACE_TOKEN), fieldNodes,
-                            null, createToken(CLOSE_BRACE_TOKEN));
-                }
-                unionType.append(recordDescriptorNode);
-            } else {
-                throw new BallerinaOpenApiException("Unsupported scenario is found. AllOf schema is given inside a " +
-                        "oneOf or anyOf schema.");
+            TypeDescriptorNode typeDescriptorNode = getTypeGenerator(schema).generateTypeDescriptorNode();
+            if (typeDescriptorNode instanceof OptionalTypeDescriptorNode &&
+                    GeneratorMetaData.getInstance().isNullable()) {
+                Node node = ((OptionalTypeDescriptorNode) typeDescriptorNode).typeDescriptor();
```

Comment:
shall we rename this `node` with the meaning full name?

Reply:
Updated to `optinalTypeDesc`.